CONTRIBUTING.md

```diff
@@ -170,3 +170,44 @@ the maintainers will decide if it is more appropriate to:
 * merge your branch,
 * squash all commit into one commit, or
 * rebase (i.e., replay) all commits on top of the master branch.
+
+
+### Legalese
+Committers are the front line gatekeepers to keep the code base clear of improperly contributed code.
+It is important to the GeoStyler users, developers and the community to avoid contributing any
+code to the project without it being clearly licensed under the project license.
+
+Generally speaking the key issues are that those providing code to be included in the repository
+understand that the code will be released under the BSD-2-Clause license, and that the person providing
+the code has the right to contribute the code. For the committer themselves understanding about
+the license is hopefully clear. For other contributors, the committer should verify the understanding
+unless the committer is very comfortable that the contributor understands the license (for
+instance frequent contributors).
+
+If the contribution was developed on behalf of an employer (on work time, as part of a work project,
+etc) then it is important that an appropriate representative of the employer understand that the
+code will be contributed under the BSD-2-Clause license. The arrangement should be cleared with an
+authorized supervisor/manager, etc.
+
+The code should be developed by the contributor, or the code should be from a source which can be
+rightfully contributed such as from the public domain, or from an open source project under a
+compatible license.
+
+All unusual situations need to be discussed and/or documented.
+
+Committer should adhere to the following guidelines, and may be personally legally liable for
+improperly contributing code to the source repository:
+
+* Make sure the contributor (and possibly employer) is aware of the contribution terms.
+* Code coming from a source other than the contributor (such as adapted from another project)
+  should be clearly marked as to the original source, copyright holders, license terms and so forth.
+  This information can be in the file headers, but should also be added to the project licensing
+  file if not exactly matching normal project licensing (LICENSE.txt).
+* Existing copyright headers and license text should never be stripped from a file. If a copyright
+  holder wishes to give up copyright they must do so in writing to the project steering committee
+  before copyright messages are removed. If license terms are changed it has to be by agreement
+  (written in email is ok) of the copyright holders.
+* When substantial contributions are added to a file (such as substantial patches) the
```

Comment:
How is `substantial` defined? Who decides?

Reply:
> How is `substantial` defined? Who decides? 

That is a good question. At first, it's the contributor who decides if a contribution is substantial by adding the name to the copyright statement. After that, I guess it should be the same people as mentioned above (admins/maintainers) who then decide if a contribution was substantial. We could add some generic points like `new feature -> substantial, bugfix -> non-substantial`, but I personally would avoid being too specific as we might run into conflicts (e.g. what happens if a bugfix covers many lines of code, whereas a feature was added with only writing a single line?). Therefore, I would rather state that such a descision is made per case.


> As far as I can see `terrestris GmbH & Co. KG` is mentioned as only copyright holder. Is that correct?

Only currently. Another option would be something like `Copyright (c) 2018-present, terrestris GmbH & Co. KG and GeoStyler contributors` as a base, as mentioned on the [osgeo-incubator-mailinglist](https://lists.osgeo.org/pipermail/incubator/2020-May/004148.html) and then add substantial contributors additionally, if wanted. But that is still open for discussion.

Reply:
Thanks for the discussion here. WRT to the copyright holders I am open for suggestions. I think these should best be done over at the linked incubator list, or should we have a separate discussion elsewhere?

Reply:
Discussing this at the incubator list is fine for me.

Reply:
Thanks for your feedback guys. I just left a little note at the incubator list regarding copyright.

Reply:
Copyroght headings have changed in https://github.com/geostyler/geostyler/pull/1272

I assume this discussion is now resolved?